store/tikv/client_test.go

```diff
@@ -19,145 +19,121 @@ import (
-		if msgID == 1 {
-			err = util.WriteMessage(conn, msgID+100, &resp)
-		} else {
-			err = util.WriteMessage(conn, msgID, &resp)
-		}
-		c.Assert(err, IsNil)
-	})
-	defer l.Close()
-	cli := newRPCClient()
-	req := &pb.Request{
-		Type: pb.MessageType_CmdGet,
-	}
-	// Wrong ID for the first request, correct for the rests.
-	_, err := cli.SendKVReq(goctx.Background(), ":61237", req, readTimeoutShort)
-	c.Assert(err, NotNil)
-	resp, err := cli.SendKVReq(goctx.Background(), ":61237", req, readTimeoutShort)
-	c.Assert(err, IsNil)
-	c.Assert(resp.GetType(), Equals, req.GetType())
+func (s *testTikvServer) KvBatchRollback(ctx goctx.Context, req *kvrpcpb.BatchRollbackRequest) (*kvrpcpb.BatchRollbackResponse, error) {
+	return &kvrpcpb.BatchRollbackResponse{}, nil
+}
+
+func (s *testTikvServer) KvScanLock(ctx goctx.Context, req *kvrpcpb.ScanLockRequest) (*kvrpcpb.ScanLockResponse, error) {
+	return &kvrpcpb.ScanLockResponse{}, nil
+}
+
+func (s *testTikvServer) KvResolveLock(ctx goctx.Context, req *kvrpcpb.ResolveLockRequest) (*kvrpcpb.ResolveLockResponse, error) {
+	return &kvrpcpb.ResolveLockResponse{}, nil
+}
+
+func (s *testTikvServer) KvGC(ctx goctx.Context, req *kvrpcpb.GCRequest) (*kvrpcpb.GCResponse, error) {
+	return &kvrpcpb.GCResponse{}, nil
+}
+
+func (s *testTikvServer) RawGet(ctx goctx.Context, req *kvrpcpb.RawGetRequest) (*kvrpcpb.RawGetResponse, error) {
+	return &kvrpcpb.RawGetResponse{}, nil
+}
+
+func (s *testTikvServer) RawPut(ctx goctx.Context, req *kvrpcpb.RawPutRequest) (*kvrpcpb.RawPutResponse, error) {
+	return &kvrpcpb.RawPutResponse{}, nil
 }
 
-func startServer(host string, c *C, handleFunc func(net.Conn, *C)) net.Listener {
-	l, err := net.Listen("tcp", host)
+func (s *testTikvServer) RawDelete(ctx goctx.Context, req *kvrpcpb.RawDeleteRequest) (*kvrpcpb.RawDeleteResponse, error) {
+	return &kvrpcpb.RawDeleteResponse{}, nil
+}
+
+func (s *testTikvServer) Coprocessor(ctx goctx.Context, req *coprocessor.Request) (*coprocessor.Response, error) {
+	return &coprocessor.Response{}, nil
+}
+
+func (s *testTikvServer) Raft(tikvpb.TiKV_RaftServer) error {
+	return nil
+}
+
+func (s *testTikvServer) Snapshot(tikvpb.TiKV_SnapshotServer) error {
+	return nil
+}
+
+func startServer(addr string, c *C, stopCh chan int) {
+	listener, err := net.Listen("tcp", addr)
 	c.Assert(err, IsNil)
-	log.Debug("Start listenning on", host)
+	log.Debug("Start listenning on", addr)
+	grpcServer := grpc.NewServer()
+	tikvpb.RegisterTiKVServer(grpcServer, &testTikvServer{})
+	go func() {
+		grpcServer.Serve(listener)
+		listener.Close()
+	}()
 	go func() {
-		for {
-			conn, err := l.Accept()
-			if err != nil {
-				return
-			}
-			go handleFunc(conn, c)
+		select {
+		case <-stopCh:
+			grpcServer.Stop()
 		}
 	}()
-	return l
+}
+
+func TestT(t *testing.T) {
+	CustomVerboseFlag = true
+	TestingT(t)
+}
+
+var _ = Suite(&testClientSuite{})
+
+type testClientSuite struct {
+}
+
+// One normally `Send`.
+func (s *testClientSuite) TestSimpleRPC(c *C) {
```

Comment:
I think we can remove the whole test suite now. It was for testing if the home-made RPC works correctly.